src/authentication/authentication-hapi-plugin.ts

```diff
@@ -38,23 +47,184 @@ class AuthenticationHapiPlugin extends HapiPlugin {
         cors: true,
       },
     }]);
+    server.route([{
+      method: 'GET',
+      path: '/signup',
+      handler: {
+        async: this.signupHandler,
+      },
+      config: {
+        bind: this,
+      },
+    }]);
+    server.route({
+      method: 'GET',
+      path: '/team-token/{teamId}',
+      handler: {
+        async: this.getTeamTokenHandler,
+      },
+      config: {
+        bind: this,
+        auth: 'admin',
+        validate: {
+          params: {
+            teamId: Joi.number().required(),
+          },
+        },
+      },
+    });
+    server.route({
+      method: 'POST',
+      path: '/team-token/{teamId}',
+      handler: {
+        async: this.createTeamTokenHandler,
+      },
+      config: {
+        bind: this,
+        auth: 'admin',
+        validate: {
+          params: {
+            teamId: Joi.number().required(),
+          },
+        },
+      },
+    });
     next();
   }
 
   public async getTeamHandler(request: Hapi.Request, reply: Hapi.IReply) {
-    const credentials = request.auth.credentials as AccessToken;
-    const user = await getUserByEmail(credentials[this.subEmailClaimKey], this.gitlab);
-    const teams = getUserTeams(user.id, this.gitlab);
-    return reply(teams);
+    try {
+      const credentials = request.auth.credentials as AccessToken;
+      // The email has been validated in validateUser at this point
+      const user = await this.gitlab.getUserByEmail(credentials[subEmailClaimKey]!);
+      const teams = await this.gitlab.getUserGroups(user.id);
+      return reply(teams[0]); // NOTE: we only support a single team for now
+    } catch (error) {
+      this.logger.error(`Can't fetch user or team`, error);
+      return reply(Boom.wrap(error, 404));
+    }
+  }
+
+  public async getTeamTokenHandler(request: Hapi.Request, reply: Hapi.IReply) {
+    try {
+      const teamId = parseInt(request.paramsArray[0], 10);
+      const teamToken: TeamToken[] = await teamTokenQuery(this.db, undefined, teamId);
+      if (!teamToken.length) {
+        throw new Error(`No token found for team ${teamId}`);
+      }
+      return reply(teamToken[0]);
+    } catch (error) {
+      return reply(Boom.notFound(error.message));
+    }
+  }
+
+  public async createTeamTokenHandler(request: Hapi.Request, reply: Hapi.IReply) {
+    try {
+      const teamId = parseInt(request.paramsArray[0], 10);
+      const teamToken = await generateAndSaveTeamToken(teamId, this.db);
+      return reply(teamToken).code(201);
+    } catch (error) {
+      return reply(Boom.badRequest(error.message));
+    }
+  }
+
+  public async signupHandler(request: Hapi.Request, reply: Hapi.IReply) {
+    let email: string | undefined;
+    let credentials: AccessToken | undefined;
+    try {
+      credentials = request.auth.credentials as AccessToken;
+      // The email has been validated in validateUser at this point
```

Comment:
Ah, is it so that `jwt` auth is turned on by default, and it needs to be overriden with `auth: false` if we don't want to use it?